### PR TITLE
Migrated button-group to CSS modules

### LINKED ISSUE #620: Selectable Button Group
# Issue Description

The selectable button group no longer toggles selected onClick. Before when selected, it had additional styling. 

## Issue Type

- [ ] New Feature
- [ ] Enhancement
- [x] Bug
- [ ] Other

Expected:
selectable-button-group

Actual: 
http://engineering.cerner.com/terra-core/#/tests/button-group-tests/selectable-button-group

### PR DESCRIPTION
**Pull request by Matt-Butler**
### Summary
* Converted CSS to be a local module
* Updated tests

Resolves #620 #549

Deployment: https://terra-core-deployed-pr-621.herokuapp.com/static/#/

@cerner/terra

button-group

### CODE REVIEW

File: packages/terra-button-group/src/ButtonGroup.jsx [hunk 1]
```diff
@@ -96,7 +98,7 @@ class ButtonGroup extends React.Component {
 
   render() {
     const { onChange, variant, size, isCompact, isSelectable, buttons, children, ...customProps } = this.props;
-    const buttonGroupClassNames = classNames(['terra-ButtonGroup',
+    const buttonGroupClassNames = cx(['buttonGroup',
```
Comment (emilyrohrbough) [inline packages/terra-button-group/src/ButtonGroup.jsx]: should this be `button-group`

Comment (Matt-Butler) [inline packages/terra-button-group/src/ButtonGroup.jsx]: it looks like progress bar used kebab case so I will update to use that.

File: packages/terra-button-group/src/ButtonGroup.jsx [hunk 2]
```diff
@@ -96,7 +98,7 @@ class ButtonGroup extends React.Component {
 
   render() {
     const { onChange, variant, size, isCompact, isSelectable, buttons, children, ...customProps } = this.props;
-    const buttonGroupClassNames = classNames(['terra-ButtonGroup',
+    const buttonGroupClassNames = cx(['button-group-button',
```
Comment (emilyrohrbough) [inline packages/terra-button-group/src/ButtonGroup.jsx]: Oops. button-group

Comment (Matt-Butler) [inline packages/terra-button-group/src/ButtonGroup.jsx]: Good catch: https://github.com/cerner/terra-core/pull/621/commits/c125da2f4d821d6ae3199e71e0f84a0257bc53d5

**Comment by PaulSolDev** [comment]
Functional review complete.  No issues identified.

**Pull request merged**